scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -763,6 +795,12 @@ def __record_warming_result(self, container_id, success=None, permanent_error=No
                     entry.blacklisted_until = None
                     entry.blacklist_reason = None
                     result_type = 'success'
+                elif already_warm:
+                    # make sure warm is set - normally redundant, but needed for tests
+                    # don't update failure counts or blacklists, as they would have been
+                    # set by a previous 'success'.
```

Comment:
Question: Don't we need to set 
```
                    entry.failure_count = 0
                    entry.blacklisted_until = None
                    entry.blacklist_reason = None
```

iiuc these entries are propagate/maintain state so the previous iteration could be a failure right? 
So my thinking is whether this iteration is a `success` (if we warmed it ourselves) or a free `already_warm` if another thread did it for us, in both cases we should be resetting state to a success case?  Am I missing something?

Reply:
I don't think we need to set them.

We *do* need to set `is_warm` so that when `update_containers_to_warm` is called (from within `record_warming_result`) the container is removed from the list of containers.

For the other values, I think it's better not to set them based on the assumption that `already_warm` is not a `success`, it's just 'something was already there' and so we leave whatever was already there - maybe it was the result of a success, maybe there was a success followed by a failure, and our `already_warm` shouldn't disturb that because it wasn't a success in its own right.